test/EndToEndTests/Tests/Client/Microsoft.OData.Client.E2E.Tests/EdmDateAndTimeOfDayTests/Tests/EdmDateAndTimeOfDayTests.cs

```diff
@@ -0,0 +1,501 @@
+                }
+
+                Assert.Equal(ODataReaderState.Completed, reader.State);
+            }
+        }
+
+        #endregion
+
+        #region Client
+
+        [Fact]
+        public async Task SelectingAnEntitysDateOrTimeOfDayPropertiesFromODataClient_WorksCorrectly()
+        {
+            // Query Property
+            var shipDate = await _context.Orders.ByKey(7).Select(o => o.ShipDate).GetValueAsync();
+            Assert.Equal(new Date(2014, 8, 31), shipDate);
+
+            var shipTime = await _context.Orders.ByKey(7).Select(o => o.ShipTime).GetValueAsync();
+            Assert.Equal(new TimeOfDay(12, 40, 05, 50), shipTime);
+        }
+
+        [Fact]
+        public async Task AProjectionSelectForDateAndTimeOfDayProperties_WorksCorrectly()
+        {
+            // Projection Select
+            var projOrder = await _context.Orders.ByKey(7).Select(o => new Common.Client.Default.Order() { ShipDate = o.ShipDate, ShipTime = o.ShipTime }).GetValueAsync();
+            Assert.True(projOrder != null);
+            Assert.Equal(new Date(2014, 8, 31), projOrder.ShipDate);
+            Assert.Equal(new TimeOfDay(12, 40, 05, 50), projOrder.ShipTime);
+        }
+
+        [Fact]
+        public async Task UpdatingDateAndTimeOfDayPropertiesFromODataClient_WorksCorrectly()
+        {
+            _context.MergeOption = MergeOption.OverwriteChanges;
+
+            // Update Properties
+            var order = await _context.Orders.ByKey(7).GetValueAsync();
+            Assert.True(order != null);
+            Assert.Equal(new Date(2014, 8, 31), order.ShipDate);
+            Assert.Equal(new TimeOfDay(12, 40, 05, 50), order.ShipTime);
+
+            order.ShipDate = new Date(2014, 9, 30);
+            _context.UpdateObject(order);
+            await _context.SaveChangesAsync();
+
+            var updatedOrder = await _context.Orders.ByKey(7).GetValueAsync();
+            Assert.Equal(new Date(2014, 9, 30), updatedOrder.ShipDate);
+        }
+
+        [Fact]
+        public async Task InvokingAFunctionThatReturnsDateFromODataClient_WorksCorrectly()
+        {
+            // Function
+            var date = await _context.Orders.ByKey(7).GetShipDate().GetValueAsync();
+            Assert.Equal(new Date(2014, 8, 31), date);
+        }
+
+        [Fact]
+        public async Task InvokingAnActionThatTakesDateAndTimeOfDayAsParameters_WorksCorrectly()
+        {
+            _context.MergeOption = MergeOption.OverwriteChanges;
+
+            var order = await _context.Orders.ByKey(7).GetValueAsync();
+            Assert.Equal(new Date(2014, 8, 31), order.ShipDate);
+
+            // Action
+            await _context.Orders.ByKey(7).ChangeShipTimeAndDate(Date.MaxValue, TimeOfDay.MaxValue).GetValueAsync();
+            order =await _context.Orders.ByKey(7).GetValueAsync();
+            Assert.Equal(Date.MaxValue, order.ShipDate);
+            Assert.Equal(TimeOfDay.MaxValue, order.ShipTime);
+        }
+
+        #endregion

```

Comment:
There would be value to add client tests for invoking `CheckShipDate` and `CheckShipTime`